### ISSUE TITLE
Optimize numerical solutiion for velocity dispersion in the generic dark matter profile class

### ISSUE THREAD

**Issue by abensonca** [galacticusorg/galacticus#138]
Solving the Jeans equation is slow. This could be improved by storing a tabulated solution (since solving the equation to find the dispersion at some radius, _r_, involves finding the solution at all radii >_r_ anyway), which is then re-used if the dispersion is requested for the same halo, and extended if a smaller radius is required.

**Comment by abensonca** [comment]
A challenge here is that the "generic" dark matter profile numerical methods can be used for multiple different darkMatterProfileDMO classes acting on the same node. For example, the dark matter profile could be built up using both "heated" and "finiteResolution" classes. So, if we store a solution to the velocity dispersion profile we need to store separate solutions for each class - in principle straightforward but we need some way to distinguish between the classes. Probably need to be able to assign each class some unique identifier which can be communicated to the generic solver class.

**Comment by abensonca** [comment]
Actually, this shouldn't be an issue providing the tables and any state is stored within the darkMatterProfileGeneric class object.

**Comment by Xiaolong-Du** [comment]
In certain cases, the default tolerance of numerical integration can not be achieved due to roundoff error. Adding a parameter that controls the numeric tolerance may be helpful.

**Comment by abensonca** [comment]
I just submitted 1536eb31f which provides for an adjustable tolerance for the velocity dispersion numerical solution in the generic dark matter profile. It's controlled by a variable `toleranceRelativeVelocityDispersion` that should be set by any `darkMatterProfile` object that makes use of the numerical solver. (Note that I set a default value of 10^-6 for this tolerance in the `darkMatterProfileGeneric` in case the child class forgets to explicitly set a value.) Currently I made the `darkMatterProfileDMOHeated` class read the value of this from a parameter - other profiles which make use of the numerical solver for velocity dispersion don't set the tolerance (so the default will be used).
